larpandoracontent/LArObjects/LArDiscreteProbabilityVector.h

```diff
@@ -0,0 +1,393 @@
+            float m_cumulativeDatum;       ///< The cumulative probability value
+            float m_width;                 ///< The width of the probability bin
+    };
+
+    typedef std::vector<DiscreteProbabilityDatum> DiscreteProbabilityData;
+
+    /**
+     *  @brief  Get a initialised probability data vector from the input data
+     *
+     *  @param  inputData the input data
+     *
+     *  @return a fully-initialised discrete probability data vector
+     */
+    template<typename TX, typename TY>
+    DiscreteProbabilityData InitialiseDiscreteProbabilityData(InputData<TX, TY> inputData) const;
+
+    /**
+     *  @brief  Get a resampled probability data vector by resampling another probability data vector
+     *
+     *  @param  discreteProbabilityVector another discrete probability vector
+     *  @param  resamplingPoints the points to resample the discrete probability vector with
+     *
+     *  @return a resampled probability data vector
+     */
+    DiscreteProbabilityData ResampleDiscreteProbabilityData(const DiscreteProbabilityVector &discreteProbabilityVector, 
+        const ResamplingPoints &resamplingPoints) const;
+
+    /**
+     *  @brief  Get a randomised probability data vector in which the x values are unchanged, the probability density is 
+     *          randomised and the cumulative probability is recalculated
+     *
+     *  @param  discreteProbabilityVector another discrete probability vector
+     *  @param  randomNumberGenerator the random number generator for the random reshuffling 
+     *
+     *  @return a resampled probability data vector
+     */
+    DiscreteProbabilityData RandomiseDiscreteProbabilityData(const DiscreteProbabilityVector &discreteProbabilityVector, 
+        std::mt19937 &randomNumberGenerator) const;
+
+    /**
+     *  @brief  Sort the input data according to their x value
+     *
+     *  @param  lhs the first InputDatum
+     *  @param  lhs the second InputDatum
+     *
+     *  @return a bool dictacting swapping the two InputDatums
+     */
+    template <typename TX, typename TY>
+    static bool SortInputDataByX(const InputDatum<TX, TY> &lhs, const InputDatum<TX, TY> &rhs);
+
+    /**
+     *  @brief  Calculate the probability normalisation
+     *
+     *  @param  inputData the input data
+     *
+     *  @return the probability normalisation
+     */
+    template <typename TX, typename TY>
+    float CalculateNormalisation(const InputData<TX, TY> &inputData) const;
+
+    /**
+     *  @brief  Verify the integrity of the complete probability vector
+     */
+    void VerifyCompleteData() const;
+
+    /**
+     *  @brief  Verify the integrity of the element request
+     *
+     *  @param  index the index in the probability vector
+     */
+    void VerifyElementRequest(const size_t index) const;
+
+    float m_xUpperBound;                                              ///< the upper bound of the probability vector
```

Comment:
We often vertically align member variable names (just cos) (as well as their accompanying comments).

Reply:
Aligned